### ISSUE TITLE
Cannot read property 'associations' of undefined

### ISSUE THREAD

**Issue by NooBiToo** [VirtusLab/strapi-plugin-navigation#28]
this exits when trying to save a menu item
```
[2021-01-26T13:46:11.416Z] error TypeError: Cannot read property 'associations' of undefined
    at W:\WWW\ds.strapi\node_modules\strapi-connector-bookshelf\lib\relations.js:259:46
    at Array.forEach (<anonymous>)
    at W:\WWW\ds.strapi\node_modules\strapi-connector-bookshelf\lib\relations.js:253:16
    at Array.reduce (<anonymous>)
    at Function.update [as updateRelations] (W:\WWW\ds.strapi\node_modules\strapi-connector-bookshelf\lib\relations.js:105:68)
From previous event:
    at W:\WWW\ds.strapi\node_modules\knex\lib\transaction.js:68:34
    at processImmediate (internal/timers.js:461:21)
From previous event:
    at new Transaction (W:\WWW\ds.strapi\node_modules\knex\lib\transaction.js:57:30)
    at new Transaction_MySQL (W:\WWW\ds.strapi\node_modules\knex\lib\dialects\mysql\transaction.js:7:1)
    at Client_MySQL.transaction (W:\WWW\ds.strapi\node_modules\knex\lib\dialects\mysql\index.js:51:12)
    at Function.transaction (W:\WWW\ds.strapi\node_modules\knex\lib\util\make-knex.js:40:31)
    at wrapTransaction (W:\WWW\ds.strapi\node_modules\strapi-connector-bookshelf\lib\queries.js:43:15)
    at Object.create (W:\WWW\ds.strapi\node_modules\strapi-connector-bookshelf\lib\queries.js:93:12)
    at fn (W:\WWW\ds.strapi\node_modules\strapi-database\lib\queries\helpers.js:31:54)
    at Object.create (W:\WWW\ds.strapi\node_modules\strapi-database\lib\queries\helpers.js:15:24)
    at async W:\WWW\ds.strapi\node_modules\strapi-plugin-navigation\services\navigation.js:509:32
    at async Promise.all (index 0)
    at async Promise.all (index 0)
```

**Comment by cyp3rius** [comment]
Hi @NooBiToo,

Can I ask for if you've provided the relation field properly into the attributes model section?

**Comment by NooBiToo** [comment]
Plugins:
```
  plugins: {
    comments: {},
    navigation: {},
  },
```
Model categories.settings.json
```
{
  "kind": "collectionType",
  "collectionName": "categories",
  "info": {
    "name": "Categories"
  },
  "options": {
    "increments": true,
    "timestamps": true
  },
  "attributes": {
    "name": {
      "type": "string"
    },
    "url": {
      "type": "uid",
      "targetField": "name"
    },
    "publish": {
      "type": "boolean"
    },
    "articles": {
      "via": "category",
      "collection": "articles"
    },
    "navigation": {
      "model": "navigationitem",
      "plugin": "navigation",
      "via": "related",
      "configurable": false,
      "hidden": true
    }
  }
}
```
Strapi: 3.1.7
Plugin: 1.0.0-beta.13

**Comment by cyp3rius** [comment]
Thanks @NooBiToo, all looks good and probably we know what the issue is. In the on-going PR #27 we've fixed a case with `collectionName` singularization. Should be merged this week.

**Comment by NooBiToo** [comment]
thank you too @cyp3rius

**Comment by cyp3rius** [comment]
@NooBiToo version `1.0.0-beta.14 ` has been just released. Issue should be solved.

**Comment by DmitryZamura** [comment]
Strapi version 3.5.1, strapi-plugin-navigation ^1.0.0-beta.15

The same error TypeError: Cannot read property 'associations' of undefined

**Comment by cyp3rius** [comment]
Thanks for raising @DmitryZamura, we've indeed identified the case and it's the same as in #35. The fix is on going. Simply there will be a change in usage of collection names.

**Comment by cyp3rius** [comment]
Ticket auto closed because of fix merge and release `1.0.0-beta.16`. Hope it fixes all the issues @NooBiToo @DmitryZamura . 

Just let me know if works, in case of on-going issue re-open please.
